TestShutdown is flaky

Becomes flaky after we start making it run in parallel.

It needs to fight for resources now, which makes it take longer.

synctest https://github.com/prometheus/prometheus/pull/17043/files will be very helpful for such "check on duration" tests.

For now we can just remove `t.Parallel()`

Waiting and monitoring so we group all the work in the same PR.

---
here https://github.com/prometheus/prometheus/actions/runs/16908464940/job/47903713137 if still active

```
--- FAIL: TestShutdown (13.23s)
    queue_manager_test.go:500: Took too long to shutdown: 1.296048814s > 1s
FAIL
```

--- a/storage/remote/queue_manager_test.go
+++ b/storage/remote/queue_manager_test.go
@@ -466,7 +466,7 @@ func TestSampleDeliveryOrder(t *testing.T) {
 }
 
 func TestShutdown(t *testing.T) {
-	t.Parallel()
+	// Not t.Parallel() because the test became flaky; see https://github.com/prometheus/prometheus/issues/17045
 	deadline := 1 * time.Second
 	c := NewTestBlockedWriteClient()
 

